tests/testthat/test-reload.R

```diff
@@ -0,0 +1,155 @@
+        test_project <- tempfile('test_project')
+        suppressMessages(create.project(test_project, minimal = FALSE))
+        on.exit(unlink(test_project, recursive = TRUE), add = TRUE)
+        
+        oldwd <- setwd(test_project)
+        on.exit(setwd(oldwd), add = TRUE)
+        
+        clear(force=TRUE)
+        
+        # Create some R code and put in data directory 
+        CODE <- paste0(deparse(substitute({
+                x <- 10
+                
+        })), collapse ="\n")
+        
+        # save R code in the data directory
+        writeLines(CODE, "data/x.R")
+        
+        CODE <- paste0(deparse(substitute({
+                y <- 20
+                
+        })), collapse ="\n")
+        
+        # save R code in the data directory
+        writeLines(CODE, "data/y.R")
+        
+        CODE <- paste0(deparse(substitute({
+                z <- 10
+                
+        })), collapse ="\n")
+        
+        # save R code in the data directory
+        writeLines(CODE, "data/z.R")
+        
+        # Read the config data and set sticky_variables to x,y
+        config <- translate.dcf("config/global.dcf")
+        expect_error(config$sticky_variables <- "x,y", NA)
+        write.dcf(config, "config/global.dcf" )
+        
+        # Load the project and check that x exists and is in the cache
+        suppressMessages(load.project())
+        expect_true(exists("x"))
+        expect_true(.read.cache.info("x")$in.cache)
+        
+        # reload the project and config, x, y should not be cleared
+        expect_message(reload.project(), "not cleared: config x y")
+        
+        # Also, z should be reloaded from data
+        expect_true(exists("z"))
+        
+        
+})
+
+test_that('reload.project with reset clears everything', {
+        test_project <- tempfile('test_project')
+        suppressMessages(create.project(test_project, minimal = FALSE))
+        on.exit(unlink(test_project, recursive = TRUE), add = TRUE)
+        
+        oldwd <- setwd(test_project)
+        on.exit(setwd(oldwd), add = TRUE)
+        
+        clear(force=TRUE)
+        
+        # Create some R code and put in data directory 
+        CODE <- paste0(deparse(substitute({
+                x <- 10
+                
+        })), collapse ="\n")
+        
+        # save R code in the data directory
+        writeLines(CODE, "data/x.R")
+        
+        CODE <- paste0(deparse(substitute({
+                y <- 20
+                
+        })), collapse ="\n")
+        
+        # save R code in the data directory
+        writeLines(CODE, "data/y.R")
+        
+        CODE <- paste0(deparse(substitute({
+                z <- 10
+                
+        })), collapse ="\n")
+        
+        # save R code in the data directory
+        writeLines(CODE, "data/z.R")
+        
+        # Read the config data and set sticky_variables to x,y
+        config <- translate.dcf("config/global.dcf")
```

Comment:
why don't you use`.read.config` and `.save.config`? I think it's a little clearer and more portable if we ever decide to use another format to store the configuration (or move the location). See also #184, where some wild variations on this were replaced with `.read.config`.